.github/workflows/podvm_mkosi.yaml

```diff
@@ -0,0 +1,92 @@
+name: Create a Pod VM image with mkosi
+
+on:
+  workflow_dispatch:
+
+jobs:
+  build-binaries:
+    name: Build binaries
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Log in to GitHub Container Registry
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build builder
+        uses: docker/build-push-action@v5
+        with:
+          tags: ghcr.io/${{ github.repository }}/podvm/builder-fedora:${{ github.sha }}
+          file: podvm/Dockerfile.podvm_builder.fedora
+          push: true
+
+      - name: Build binaries
+        uses: docker/build-push-action@v5
+        with:
+          tags: ghcr.io/${{ github.repository }}/podvm/binaries-fedora:${{ github.sha }}
+          file: podvm/Dockerfile.podvm_binaries.fedora
+          push: true
+          build-args:
+            "BUILDER_IMG=ghcr.io/${{ github.repository }}/podvm/builder-fedora:${{ github.sha }}"
+
+  build-image:
+    name: Build image
+    needs: [build-binaries]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Install Nix
+        uses: cachix/install-nix-action@v22
+
+      - name: Build nix shell to cache dependencies
+        run: nix build .#devShells.x86_64-linux.podvm-mkosi
+
+      - name: Install crane
+        run: nix profile install nixpkgs#crane
+
+        # This removes the checkout and creates a btrfs volume with maximized
+        # build space.
+      - name: Maximize build space
```

Comment:
I suspect this is due to limited space available on the builder for images. Or why do we need that? locally my build folder is rather small (2GB)

Reply:
> Are you building on ext4 locally?
yup, maybe we can take a look at the mkosi CI jobs as a reference?

Reply:
The images they build in their CI are quite minimal. There aren't any actions regarding disk size, so I don't think it is a problem for them. We have the same space problem in Constellation. I would like to investigate at some point, but I couldn't find the cycles yet to dig into it.